packages/docs/Examples/sale.md

```diff
@@ -5,120 +5,244 @@ icon: flame
 
 # Sales
 
-In the case of a swap, both smart objects have to be known before the swap transaction can be built. In this section we show how a single smart object can be offered for sale at a specified price.
+!!!
+These examples use several advanced features (sighash types, mocking, and controlling the order of inputs and outputs, ordinal safe programming) that are not sufficiently documented elsewhere. If you would like to use these features we suggest to ask about their safe use [here](https://t.me/thebitcoincomputer).
+!!!
 
-## Sale
+In a swap, the smart objects to be swapped are known before the swap transaction is built. In this section we show how a single smart object can be offered for sale at a specified price. Unlike in the case of a swap where both objects are known in advance, the output that will be provided by the buyer later is not known when the swap transaction is built. The Bitcoin Computer provides a feature called mocking that Seller can use to mock up the payment object that Buyer will create later.
 
-We first explain how to execute the sale of a smart object. The smart contract is quite simple but the swap does not preserving ordinal ranges of the smart object being sold. It is therefore not safe to use with ordinals. In the next section we explain a slightly more complicated swap that can be used with ordinals. 
+## Smart Object Sale
 
-### Smart Contract
+We first explain how to execute the sale of a smart object. The smart contract is quite simple but does not preserve the ordinal ranges of the smart object being sold. It is therefore not safe to use with ordinals. The [next](#ordinals-sale) section explains a slightly more complicated swap that can be used with ordinals.
 
-```ts
-class Sale extends Contract {
-  static exec(a: NFT, b: NFT) {
-    const [ownerA] = a._owners
-    const [ownerB] = b._owners
-    a.transfer(ownerB)
-    b.transfer(ownerA)
-    return [b, a]
-  }
-}
-```
+The idea is to build a "crossover" transaction with two inputs and two outputs where the first inputs spends the NFT `n`, the second output spends a payment smart object `p`, the first output is `p` after the swap and the second output is `n` after the swap. 
+
+Seller signs the first input and output with the sighash type `SIGHASH_SINGLE | SIGHASH_ANYONECANPAY` so that any other user can modify the second input and output or add arbitrary inputs and outputs as longs as the input-output pair signed by Seller have the same index. As the input-output pair is signed, Seller is guaranteed that if a transaction containing the pair is included in the blockchain then Seller will get paid. 
```

Comment:
Seller signs the first input and output with the [sighash type](https://developer.bitcoin.org/devguide/transactions.html#signature-hash-types) `SIGHASH_SINGLE | SIGHASH_ANYONECANPAY` so that any other user can modify the second input and output or add arbitrary inputs and outputs. The important aspect is that the input-output pair signed by the Seller must have the same index, in this case, the first index. As the input-output pair is signed, the Seller is guaranteed that if a transaction containing the pair is included in the blockchain, then the Seller will get paid.